tools/serverpod_cli/lib/src/generator/dart/library_generators/library_generator.dart

```diff
@@ -1213,6 +1208,17 @@ extension TypeDefinitionReturnsRecordInContainer on TypeDefinition {
             (g.isMapType || g.isListType || g.isSetType) &&
                 g.returnsRecordInContainer));
   }
+
+  /// Returns `true` if the type is or contains a non-String-keyed Map in a container (non-model) type
+  bool get containsNonStringKeyedMap {
+    return (isMapType && generics.first.className != 'String') ||
+        ((isMapType || isListType || isSetType) &&
+            generics.any((g) =>
+                (g.isMapType && g.generics.first.className != 'String') ||
+                // Important to only check default container types, there is not need to descent into model classes
+                (g.isMapType || g.isListType || g.isSetType) &&
```

Comment:
Suggestion: I think it would make more sense to check if there are any generics instead as that is more robust incase we would add ay other container classes. Also we still need to check nested records too right? 

We could have something like where the key is a record with a Map that has a model as a key.
```dart
(Map<(Map<MyModel, String>, String), String>)
```

^ probably a good test case to add!

Reply:
Nit: ... there is **no** need to descent ...